### ISSUE TITLE
some questions regarding the results

### ISSUE THREAD

**Issue by bpwl0121** [hoangthangta/ThangKAN#3]
Hi,

from the results you showed in the README, I found the test acc are almost same among EfficientKAN, TransformerMLP etc. you use KAN with few layers, but TransformerMLP with bert backbone, is KAN so powerful? or you just train for 10 epoch which is insufficient

thanks

**Comment by bpwl0121** [comment]
![image](https://github.com/hoangthangta/ThangKAN/assets/38959389/609590f7-8bfb-4b98-b783-1272f2fcb6ba)
btw, your progressbar is quite misleading....

**Comment by bpwl0121** [comment]
![image](https://github.com/hoangthangta/ThangKAN/assets/38959389/015e8164-b22f-45fa-b867-255be0ca75b5)
I tried efficientkan and mlp on mrpc task. I guess the the task is too simple, and value is quickly saturated.

**Comment by hoangthangta** [comment]
Thank you for your interesting question. This is due to the nature of the GLUE datasets. So what is the difference? It is training time and model storage. I did not try with other model embeddings or even weights, last hidden layer.

**Comment by bpwl0121** [comment]
I mean some tasks like mrpc achieve around 0.67 easliy even if I just train 1 epoch, I guess. have you ever tried embedding+2 layers MLP for glue dataset?

**Comment by hoangthangta** [comment]
For MRPC, the training on "last_hidden_state" with input size 512 x 768 = 393216 will produce an accuracy ~ 73% after 10 epochs.

```python run_train.py --mode "train" --network "efficientkan" --em_model_name "bert-base-cased" --ds_name "wnli" --epochs 10 --batch_size 4 --max_len 512 --n_size 512 --m_size 768 --n_hidden 64 --n_class 2```

```if (embed_type == 'hidden'): # last hidden state, 512 x 768 for BERT
            outputs = em_model(input_ids=input_ids, attention_mask=attention_mask)
            embeddings = outputs['last_hidden_state']
            #embeddings = torch.sum(embeddings, (1), keepdim = True) # require memory 
            embeddings = embeddings.view(-1, n_size*m_size)```

**Comment by hoangthangta** [comment]
I updated the code and the problem is solved.
